### PR TITLE
sentry-native: init at 0.4.11

### PR DESCRIPTION
**Pull request by wheelsandmetal**
###### Motivation for this change
Add sentry-native lib

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/sentry-native/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+stdenv.mkDerivation rec {
+  pname = "sentry-native";
+  version = "0.4.11";
+
+  src = fetchFromGitHub {
+    owner = "getsentry";
+    repo = "sentry-native";
+    rev = "${version}";
+    sha256 = "sha256-RmCFpqm82lTh1AHxfTQRsnxTGOw3JowECiLo3pc0/rE=";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/sentry-native/default.nix:12]: Please devendor the libs.

Comment (wheelsandmetal) [reply pkgs/development/libraries/sentry-native/default.nix:12]: Hey, not sure exactly what you mean by this?

Do you mean to get the src files from somewhere else? If so could you point me to an example?

Comment (SuperSandro2000) [reply pkgs/development/libraries/sentry-native/default.nix:12]: We do not want to use the vendored libs. We want to the ones from the package manager to only have one version get updates and security fixes and have nixos specific patches.

Try not fetching the submodules and add the packages to buildInputs if the build script picks them up. If not you can symlink them to the expected locations. If some packages are not in nixpkgs they should be packaged.

File: pkgs/development/libraries/sentry-native/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ lib, stdenv, fetchFromGitHub, cmake, curl, breakpad }:
+
+stdenv.mkDerivation rec {
+  pname = "sentry-native";
+  version = "0.4.11";
+
+  src = fetchFromGitHub {
+    owner = "getsentry";
+    repo = "sentry-native";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/sentry-native/default.nix:10]:
```suggestion
    rev = version;
```

File: pkgs/development/libraries/sentry-native/default.nix (hunk 1 above)
Comment (wheelsandmetal) [reply pkgs/development/libraries/sentry-native/default.nix]: Ah thanks! Didn't know about this

While digging around in the cmake flags I found one that would use the system libs for breakpad, which is even better I guess!

**Pull request merged**